fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1810,30 +1856,15 @@ private void updateIndexState(@Nonnull String indexName, byte[] indexKey, @Nonnu
             return preloadRecordStoreStateAsync().thenCompose(vignore -> markIndexNotReadable(indexName, indexState));
         }
 
+        CompletableFuture<Boolean> result = AsyncUtil.READY_FALSE;
         addIndexStateReadConflict(indexName);
-
         beginRecordStoreStateWrite();
-        boolean haveFuture = false;
-        try {
-            // A read is done before the write in order to avoid having unnecessary
-            // updates cause spurious not_committed errors.
-            byte[] indexKey = indexStateSubspace().pack(indexName);
-            Transaction tr = context.ensureActive();
-            CompletableFuture<Boolean> future = tr.get(indexKey).thenApply(previous -> {
-                if (previous == null || !Tuple.fromBytes(previous).get(0).equals(indexState.code())) {
-                    updateIndexState(indexName, indexKey, indexState);
-                    return true;
-                } else {
-                    return false;
-                }
-            }).whenComplete((b, t) -> endRecordStoreStateWrite());
-            haveFuture = true;
-            return future;
-        } finally {
-            if (!haveFuture) {
-                endRecordStoreStateWrite();
-            }
+        if (!recordStoreStateRef.get().getState(indexName).equals(indexState)) {
+            updateIndexState(getRecordMetaData().getIndex(indexName), indexState);
+            result = AsyncUtil.READY_TRUE;
         }
+        endRecordStoreStateWrite();
```

Comment:
I was thinking to drop it, because it's done in `updateIndexState`. But we would need it because of the `getState` call before it. sure, let me do what you are suggesting.